sdk/communication/azure-communication-sms/src/samples/java/com/azure/communication/sms/samples/quickstart/ReadmeSamples.java

```diff
@@ -111,4 +203,74 @@ public void sendSMSTroubleshooting() {
             System.out.println(ex.getMessage());
         }
     }
+
+    public void catchHttpErrorOnRequestAsync() {
```

Comment:
I dont think we need to both async and sync versions. It creates a lot of duplication